src/marqo/core/index_management/index_management.py

```diff
@@ -311,6 +329,10 @@ def _get_vespa_application(self, check_configured: bool = True, need_binary_file
 
         if need_binary_file_support and vespa_version < self._MINIMUM_VESPA_VERSION_TO_SUPPORT_UPLOAD_BINARY_FILES:
             # Binary files are only supported using VespaApplicationFileStore prior to Vespa version 8.382.22
+
+            # Please note that we don't pass in `check_for_application_convergence` parameter because we will always
+            # wait for convergence in this case. This will reduce the risk of overriding a newer version of the Vespa
+            # application package.
             application_package_store = VespaApplicationFileStore(
```

Comment:
just curious, are we not checking convergence for this case due to our design decision? https://www.notion.so/Skip-convergence-check-when-bootstrapping-rolling-back-Marqo-0c65ef5ab70b44ee992665f0ab2cd482?pvs=4#13d75d43da4c80388506f913eeb4c272

Reply:
Yes, correct. Old version of Vespa does not support optimistic locking mechanism for bootstrapping, which is more likely to cause the newer version of app be overridden if we do not wait for convergence.

Reply:
@adityabharadwaj198 the opposite actually, in this case we're always checking convergence